src/hotspot/share/opto/superword.cpp

```diff
@@ -3970,20 +3986,24 @@ bool SWPointer::offset_plus_k(Node* n, bool negate) {
       return true;
     }
   }
-  if (invariant(n)) {
+  if (!is_main_loop_member(n)) {
```

Comment:
Please, add comment here explaining why !is_main_loop_member() is used in this case instead of invariant().
Other places looks good.

Reply:
I updated it and also moved the following `invariant()` check into the if block as it can only hold if `!is_main_loop_member()` is true.